### ISSUE TITLE
Error during the deployment but the GA pass

### ISSUE THREAD

**Issue by GerardPaligot** [google-github-actions/deploy-cloud-functions#71]
### TL;DR

When I'm trying to publish a cloud function trigger by HTTP request, the GitHub Actions pass but the cloud function isn't updated on GCP/FirebasE.

**Expected behavior**

If there is any error during the deployment, the GA shouldn't pass and give me information to know how to fix the deployment.

**Observed behavior**

Give this output in the log of the GitHub Action:

```
Function deployment updated
No URL set. Only HttpsTrigger Cloud Functions have URL.
```

### Reproduction

**Action YAML**

```yaml
name: Deploy

on:
  push:
    branches:
      - main

jobs:
  deploy-functions:
    runs-on: latest-ubuntu
    env:
      working-directory: ./functions
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2.1.2
        with:
            node-version: '12'
      - uses: google-github-actions/setup-gcloud@master
        with:
          project_id: ${{ secrets.GCP_PROJECT_ID }}
          service_account_key: ${{ secrets.SERVICE_ACCOUNT_KEY }}
          export_default_credentials: true
      - run: npm i -g firebase-tools
      - run: yarn
        working-directory: ${{ env.working-directory }}
      - uses: google-github-actions/deploy-cloud-functions@v0.5.0
        with:
          name: api # here a cloud function trigger by a http request
          runtime: nodejs12
          credentials: ${{ secrets.SERVICE_ACCOUNT_KEY }}
      - uses: google-github-actions/deploy-cloud-functions@v0.5.0
        with:
          name: uploadCourses # here a cloud function trigger by an event on Google Storage
          runtime: nodejs12
          credentials: ${{ secrets.SERVICE_ACCOUNT_KEY }}
```

**Additional information**

Maybe linked to https://github.com/google-github-actions/deploy-cloud-functions/issues/47?

It seems, there is an error in the archive uploaded on GCP.

<img width="1033" alt="Capture d’écran 2021-06-15 à 11 38 36" src="https://user-images.githubusercontent.com/1913901/122030628-3c168d00-cdce-11eb-8976-3471ff39c4ef.png">

**Comment by bharathkkb** [comment]
Thanks for the report @GerardPaligot 
Glancing at the logs, the error seems to be coming during the function execution so I am not sure if it is due to an error in the archive. If the repo where this occurred is public, could you link it? It would be super helpful to repro

**Comment by sullerandras** [comment]
We have the same issue, the Github Actions passes but the function is not deployed. In the web interface it says "Function is active, but the last deploy failed".

**Comment by bharathkkb** [comment]
@sullerandras if your repo is public, it would be helpful for a repro. In the above screenshot the error seemed to occur during function execution but looks like in your case deploy is failing?

**Comment by sullerandras** [comment]
@bharathkkb unfortunately our repo is also private. In our case the function's the docker image was built successfully but it raised an error during the deployment because the `init` function `panic`'ed.
For example we read the secret in the `init` function and it did not have permission to access the secret (which results in a `panic`). In this case the cloud function ends up in the "Function is active, but the last deploy failed" state and the github actions build is green, and we can only see the error in the function's logs (and the cloud function's previous version stayed active).
If we try to deploy the same function using gcloud cli then it fails and prints an error message there.

**Comment by bharathkkb** [comment]
@sullerandras thanks for the detailed info. I'll try to replicate this and see if we can add some additional checks. We currently only [poll the the create op](https://github.com/google-github-actions/deploy-cloud-functions/blob/d1590b351306028e9fd804189cb0cca0e522a488/src/cloudFunctionClient.ts#L283-L288) to completion.
